tinygrad/tensor.py

```diff
@@ -108,9 +108,27 @@ def randn(*shape):
   def eye(dim):
     return Tensor(np.eye(dim).astype(np.float32))
 
+  def zero_grads(self):
+    if self.grad is not None:
+      self.grad = None
+      if self._ctx:
+        for parent in self._ctx.parents:
+          parent.zero_grads()
+
+  def deepwalk(self, visited=None, a=None):
```

Comment:
Put this in the backward function, check out how micrograd does it:

https://github.com/karpathy/micrograd/blob/master/micrograd/engine.py#L54